test/ngAnimate/animateRunnerSpec.js

```diff
@@ -0,0 +1,251 @@
+
+    inject(function($animateRunner) {
+      var host = {};
+      var spy = host[method] = jasmine.createSpy();
+      var runner = new $animateRunner(host);
+      runner[method]();
+      expect(spy).toHaveBeenCalled();
+    });
+  });
+
+  they("should trigger the inner runner's host $prop function",
+    ['end', 'cancel', 'pause', 'resume'], function(method) {
+
+    inject(function($animateRunner) {
+      var host = {};
+      var spy = host[method] = jasmine.createSpy();
+      var runner1 = new $animateRunner();
+      var runner2 = new $animateRunner(host);
+      runner1.setHost(runner2);
+      runner1[method]();
+      expect(spy).toHaveBeenCalled();
+    });
+  });
+
+  it("should resolve the done function only if one RAF has passed",
+    inject(function($animateRunner, $$rAF) {
+
+    var runner = new $animateRunner();
+    var spy = jasmine.createSpy();
+    runner.done(spy);
+    runner.complete(true);
+    expect(spy).not.toHaveBeenCalled();
+    $$rAF.flush();
+    expect(spy).toHaveBeenCalled();
+  }));
+
+  it("should resolve with the status provided in the completion function",
+    inject(function($animateRunner, $$rAF) {
+
+    var runner = new $animateRunner();
+    var capturedValue;
+    runner.done(function(val) {
+      capturedValue = val;
+    });
+    runner.complete('special value');
+    $$rAF.flush();
+    expect(capturedValue).toBe('special value');
+  }));
+
+  they("should immediately resolve each combined runner in depth-first order when $prop is called",
+    ['end', 'cancel'], function(method) {
+
+    inject(function($animateRunner, $$rAF) {
+      var runner1 = new $animateRunner();
+      var runner2 = new $animateRunner();
+      runner1.setHost(runner2);
+
+      var status1, status2, signature = '';
+      runner1.done(function(status) {
+        signature += '1';
+        status1 = status;
+      });
+
+      runner2.done(function(status) {
+        signature += '2';
+        status2 = status;
+      });
+
+      runner1[method]();
+
+      var expectedStatus = method === 'end' ? true : false;
+      expect(status1).toBe(expectedStatus);
+      expect(status2).toBe(expectedStatus);
+      expect(signature).toBe('21');
+    });
+  });
+
+  they("should resolve/reject using a newly created promise when .then() is used upon $prop",
+    ['end', 'cancel'], function(method) {
+
+    inject(function($animateRunner, $rootScope) {
+      var runner1 = new $animateRunner();
+      var runner2 = new $animateRunner();
+      runner1.setHost(runner2);
+
+      var status1;
+      runner2.then(
+        function() { status1 = 'pass'; },
+        function() { status1 = 'fail'; });
+
+      var status2;
+      runner2.then(
+        function() { status2 = 'pass'; },
+        function() { status2 = 'fail'; });
+
+      runner1[method]();
+
+      var expectedStatus = method === 'end' ? 'pass' : 'fail';
+
+      expect(status1).toBeFalsy();
+      expect(status2).toBeFalsy();
```

Comment:
Aren't they expected to be undefined ? If so, using `toBeUndefined()` would be easier to follow.
(`toBeFalsy()` had me wondering if I had missed something.)